File: Java-Advanced/src/main/java/com/gs/fiap/jobfitscore/domain/curso/Curso.java
package com.gs.fiap.jobfitscore.domain.curso;

import com.gs.fiap.jobfitscore.domain.usuario.Usuario;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "cursos")
public class Curso {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long idCurso;
	
	@Column(nullable = false, length = 150)
	private String nome;
	
	@Column(length = 150)
	private String instituicao;
	
	private Integer cargaHoraria;
	
	@ManyToOne(optional = false)
	@JoinColumn(name = "usuario_id", nullable = false)
	private Usuario usuario;
}

File: Advanced-Business-Development-With -DotNET/Models/Curso.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace JobFitScoreAPI.Models
{
    [Table("CURSOS")]
    public class Curso
    {
        [Key]
        [Column("id_curso")]
        public int IdCurso { get; set; }

        [Required]
        [Column("nome")]
        [MaxLength(150)]
        public string Nome { get; set; } = string.Empty;

        [Column("instituicao")]
        [MaxLength(150)]
        public string? Instituicao { get; set; }

        [Column("carga_horaria")]
        public int? CargaHoraria { get; set; }

        [Column("data_conclusao")]
        public DateTime? DataConclusao { get; set; }

        [Column("descricao")]
        [MaxLength(500)]
        public string? Descricao { get; set; }

        [Required]
        [Column("usuario_id")]
        public int UsuarioId { get; set; }

        
        public Usuario? Usuario { get; set; }
    }
}
